File: updated_code/code/java/1512-number-of-good-pairs.java
package com.lloydcode.quality;

/**
 * Solution for the "Number of Good Pairs" problem.
 */
public class Solution {

    /**
     * Returns the number of good pairs in the given array of integers.
     * 
     * @param nums an array of integers
     * @return the number of good pairs
     */
    public int numIdenticalPairs(int[] nums) {
        int count = 0;
        final Map<Integer, Integer> map = new HashMap<>();
        int num;
        for (int i = 0; i < nums.length; i++) {
            num = nums[i];
            if (map.containsKey(num)) {
                count += map.get(num);
            }
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        return count;
    }
}

File: updated_code/code/python/1512-number-of-good-pairs.py
Python
from typing import List

class Solution:

    """
    This class contains a method numIdenticalPairs that calculates the number of good pairs in an array of integers.
    
    A good pair is a pair of integers (i, j) that have the same value and i < j.
    
    Parameters:
    nums (List[int]): The list of integers.
    
    Returns:
    int: The number of good pairs.
    """

    """
    Calculate the number of good pairs in an array of integers.
    
    Parameters:
    nums (List[int]): The list of integers.
    
    Returns:
    int: The number of good pairs.
    """
    def numIdenticalPairs(self, nums: List[int]) -> int:
        """
        Calculate the number of good pairs in an array of integers.
    
        Parameters:
        nums (List[int]): The list of integers.
    
        Returns:
        int: The number of good pairs.
        """
        count = 0
        for i, num in enumerate(nums):
            for j in range(i + 1, len(nums)):
                if num == nums[j]:
                    count += 1
        return count